src/world/world_generator.rs

```diff
@@ -0,0 +1,174 @@
+/*
+    [
+        (x0, y0) (x1, y0)...
+        (x0, y1) (x1, y1)...
+        ...
+    ]
+*/
+use rand::{Rng, SeedableRng};
+use crate::world::World;
+use crate::world::block_textures::BlockTypes;
+
+const MAP_WIDTH: u32 = World::MAP_WIDTH;
+const MAP_HEIGHT: u32 = World::MAP_HEIGHT;
+const MAP_AREA: u32 = MAP_WIDTH * MAP_HEIGHT;
+
+pub fn generate_world() -> Box<[u16; MAP_AREA as usize]> {
+    const SEED: [u8; 32] = [1; 32];
+
+    let mut rng: rand::rngs::StdRng = rand::rngs::StdRng::from_seed(SEED);
+
+    let mut world_map: Box<[u16; MAP_AREA as usize]> = Box::new([0u16; MAP_AREA as usize]);
+
+    fn set_block(world_map: &mut Box<[u16; MAP_AREA as usize]>, x: u32, y: u32, block_type: BlockTypes) {
+        match world_map.get((x + y * MAP_WIDTH) as usize) {
+            Some(_) => world_map[(x + y * MAP_WIDTH) as usize] = block_type as u16,
+            None => {
+                println!("Error: Write to out of bounds coordinate X:{}, Y:{}", x, y);
+            },
+        }
+    }
+
+    fn get_block(world_map: &Box<[u16; MAP_AREA as usize]>, x: u32, y: u32) -> Result<u16, ()> {
+        match world_map.get((x + y * MAP_WIDTH) as usize) {
+            Some(&result) => Ok(result),
+            None => {
+                println!("Error: Read to out of bounds coordinate X:{}, Y:{}", x, y);
+                Err(())
+            },
+        }
+    }
```

Comment:
Seems like this was useful at the start, but I think we should get rid of the error handling and make it panic on out of bounds (since that should never happen for final-ish code). Also looks like the Result (for set) is unused anyway